cmd/get_logs.go

```diff
@@ -167,31 +172,43 @@ func (glc *getLogsCmd) run() (err error) {
 	if err = glc.getClusterNodes(); err != nil {
 		return errors.Wrap(err, "listing cluster nodes")
 	}
-	for _, n := range glc.masterNodes {
-		log.Infof("Processing master node: %s\n", n.Name)
-		out, err := glc.collectLogs(n, glc.linuxSSHConfig)
-		if err != nil {
-			log.Warnf("Remote command output: %s", out)
-			log.Warnf("Error: %s", err)
+	if glc.linuxScriptPath == "" && (!glc.cs.Properties.MasterProfile.IsVHDDistro()) {
+		log.Warnf("No linux log collecting script provided, will skip log collection for master nodes as distro is not aks VHD")
```

Comment:
I think this may be cleaner if `getClusterNodes` is in charge of controlling which nodes make it into the nodes collections. Quick pseudo func that you may need:

```
func linuxScriptAvailable() bool { 
  return linuxScriptPath != "" || isVHDDistro()
}
```

Reply:
updated